ynr/apps/uk_results/management/commands/import_2010_parl_results.py

```diff
@@ -0,0 +1,85 @@
+from django.core.management.base import BaseCommand
+from django.contrib.postgres.search import SearchQuery, SearchVector
+from slugify import slugify
+from candidates.models.popolo_extra import Ballot
+from django.db.models.functions import Length
+from popolo.models import Membership
+
+from uk_results.helpers import read_csv_from_url
+from uk_results import models
+
+
+class Command(BaseCommand):
+
+    EDGECASES = {
+        "ynys mon": "ynys môn",
+        "down south": "south down",
+        "down north": "north down",
+    }
+
+    def find_ballot(self, seat):
+        try:
+            return self.qs.get(post__label__iexact=seat)
+        except Ballot.DoesNotExist:
+            pass
+
+        matches = self.qs.filter(post__label__search=seat)
+        try:
+            return matches.get()
+        except Ballot.MultipleObjectsReturned:
+            return matches.annotate(label_len=Length("post__label")).get(
+                label_len=len(seat)
+            )
+
+    def find_candidacy(self, ballot, name):
+        # exact search first
+        try:
+            return ballot.membership_set.get(person__name__iexact=name)
+        except Membership.DoesNotExist:
+            pass
+
+        # fallback to a smarter searching
+        vector = SearchVector("person__name", "person__other_names__name")
+        candidates = ballot.membership_set.annotate(names=vector)
+        try:
+            return candidates.get(names=name)
+        except (Membership.DoesNotExist, Membership.MultipleObjectsReturned):
+            pass
+
+        # get less specific - split the name and try and find a single match where at
+        # least one of the names match
```

Comment:
It might be a good idea to filter / check on party here?

I've found before that, especially for general elections where there's only one candidate per party, matching on party first and then a fuzzy check on the name is fairly reliable.